blottertrax/helper/title_parser.py

```diff
@@ -1,53 +1,140 @@
-import re
-
-from blottertrax.value_objects.parsed_submission import ParsedSubmission
-
-
-class TitleParser:
-
-    @staticmethod
-    def create_parsed_submission_from_submission(submission) -> ParsedSubmission:
-        artist = None
-        feature_artist = None
-        song_title = None
-        post_title = submission.title
-
-        for dash in [' - ', ' — ', ' -- ', ' - - ', ' —— ']:
-            if dash in post_title:
-                split_result = post_title.split(dash)
-                artist = split_result[0]
-                song_title = split_result[1]
-
-                break
-
-        for feature in [' feat. ', ' ft. ', ' feature ', ' featuring ', ' & ']:
-            if feature not in post_title:
-                continue
-
-            feat_index = post_title.index(feature)
-
-            # remove featuring from artist if exists
-            if len(artist) > feat_index:
-                artist = artist[:feat_index].strip()
-
-            feat_index += len(feature)
-
-            # isolate featuring artist
-            feature_artist = post_title[feat_index:].strip()
-
-            for end_char in [' -', ')', '[', ' —']:
-                if end_char in feature_artist:
-                    split_result = feature_artist.split(end_char)
-                    feature_artist, song_title = split_result
-
-                    break
-
-        # Remove everything between () and []
-        song_title = re.sub(r"\(.*\)", "", song_title)
-        song_title = re.sub(r"[.*]", "", song_title)
-        song_title = song_title.strip()
-
-        if artist is not None:
-            return ParsedSubmission(artist, feature_artist, song_title, submission.url)
-
-        raise LookupError
+from blottertrax.value_objects.parsed_submission import ParsedSubmission
+
+class TitleParser:
+
+    # Calls the main parsing methods and returns a parsed submission if successful.
+    @classmethod
+    def create_parsed_submission_from_submission(self, submission) -> ParsedSubmission:
+        artist = self.get_artist_name_from_submission_title(submission.title)
+        if artist[0] is not None:
+            song = self.get_song_name_from_submission_title(submission.title, artist)
+            return ParsedSubmission(artist[0], artist[1], song, submission.url) #, submission.id)
+
+        raise LookupError
+
+    # If featuring artist exists, returns featuring artist, else returns main artist.
+    @staticmethod
+    def get_prioritized_artist(parsed_submission):
+        if parsed_submission.featuring_artist is None:
```

Comment:
Always returns the same value?

return parsed_submission.artist

Reply:
That's a typo, should be featuring artist for the part in the if. Probably a bad merge or something. I'll get that fixed tomorrow

Reply:
Should the "featured" artist ever be prioritized over the main artist?

Also, I don't think this function is actually called anywhere.

Reply:
I wrote this because when I made this code the code was using the prioritized artist in several spots, and I mainly wrote my code to have as little effect on that as possible so I implemented a function for it. If that is no longer needed for other code I can remove. It's not really necessary for anything relating to the repost checking.

Reply:
Think we can drop this block.  I suspect if something was using the featured artist over the main artist it was likely an accident.